make_prg/subcommands/from_msa.py

```diff
@@ -52,95 +75,94 @@ def register_parser(subparsers):
+    prefix = str(temp_dir / msa_name)
 
-    prg_fname = ofile_prefix.with_suffix(".prg")
-    if prg_fname.exists() and options.no_overwrite:
-        logging.info(f"Re-using existing prg file {prg_fname}")
-        aseq = prg_builder.PrgBuilder(
-            options.MSA,
+    try:
+        builder = prg_builder.PrgBuilder(
+            locus_name=locus_name,
+            msa_file=msa_filepath,
             alignment_format=options.alignment_format,
             max_nesting=options.max_nesting,
-            min_match_length=options.min_match_length,
-            prg_file=prg_fname,
+            min_match_length=options.min_match_length
         )
+
+        logger.info(f"Writing output files of locus {msa_name}")
+        prg = builder.build_prg()
+        builder.write_prg(prefix, prg)
+        gfa.GFA_Output.write_gfa(prefix, prg)
+        builder.serialize(f"{prefix}.pickle")
+        if options.output_graphs:
+            builder.output_debug_graphs(Path(options.output_prefix + "_debug_graphs"))
+
+    except ValueError as value_error:
+        if "No records found in handle" in value_error.args[0]:
+            logger.warning(f"No records found in MSA {msa_filepath}, skipping...")
+        else:
+            raise value_error
+
+
+def run(cl_options):
+    global options
+    options = cl_options
+
+    logger.info("Getting input files...")
+    input_files = get_all_input_files(options.input)
+
+    there_is_no_input_files = len(input_files) == 0
+    if there_is_no_input_files:
+        raise FileNotFoundError(f"No input files found in {options.input}")
+
+    is_a_single_file = len(input_files) == 1
+    if is_a_single_file:
+        logger.info("A single file was given as input, single outputs will be produced")
     else:
-        aseq = prg_builder.PrgBuilder(
-            options.MSA,
-            alignment_format=options.alignment_format,
-            max_nesting=options.max_nesting,
-            min_match_length=options.min_match_length,
-        )
-        logging.info(f"Write PRG file to {prg_fname}")
-        io_utils.write_prg(prg_fname, aseq.prg, options)
-        m = aseq.max_nesting_level_reached
-        logging.info(f"Max_nesting_reached\t{m}")
+        logger.info("Multiple files were given as input, multiple outputs will be produced")
+
+    if io_utils.output_files_already_exist(options.output_prefix):
+        raise RuntimeError("One or more output files already exists, aborting run...")
+
+    output_dir = Path(options.output_prefix).parent
+    output_dir.mkdir(parents=True, exist_ok=True)
+    temp_dir = io_utils.create_temp_dir(output_dir)
+    options.temp_dir = temp_dir
 
-    if options.output_type.gfa:
-        gfa_fname = ofile_prefix.with_suffix(".gfa")
-        logging.info(f"Write GFA file to {gfa_fname}")
-        io_utils.write_gfa(gfa_fname, aseq.prg)
+    logger.info(f"Using {options.threads} threads to generate PRGs...")
+    with multiprocessing.Pool(options.threads, maxtasksperchild=1) as pool:
```

Comment:
Could we add some functionality such that if `-t 0` then use use `None`, which in turn uses all available?

Reply:
Done in https://github.com/iqbal-lab-org/make_prg/pull/35/commits/647a1071ff246be11284531d092040ce810cba6b in a slightly different manner so that we can print the actual number of threads we are using in the log